Detailed upgrade playground instructions (#3354)
PostHog/posthog.com: contents/handbook/engineering/release-new-version.md

- [ ] Upgrade PostHog playground
    - The PostHog Playground uses a helm chart deployment on Digital Ocean. Find the playground cluster in our [Digital Ocean kubernetes clusters list](https://cloud.digitalocean.com/kubernetes/clusters?i=7cfa7c).
    - If this is your first time on Digital Ocean, you'll see the below screen. If it's not, or you don't see the Getting Started flow, click "Remind me how to use this file to connect to the cluster" in the "Config file" section under the "Overview" tab. Click Get Started.

  ![PostHog - Get Started Kubernetes](../../images/05/digital_ocean_release_01.png)

    - Copy the automatic connection script by clicking the copy icon.
  
  ![PostHog - Copy Script Kubernetes](../../images/05/digital_ocean_release_02.png)

    - Open terminal and run the command you copied. This command will set the correct kubectl context for the playground environment. As a sanity check, run `kubectl config current-context` and make sure that the current context name has `playground` in it somewhere.

>     - Open another terminal window and run `k9s`. Use the arrow keys to scroll down to the posthog clusters and keep an eye on this for the duration of the upgrade. [`k9s`](https://k9scli.io/) is a terminal GUI that makes it easier to manage and observe your deployed Kubernetes applications.

Reviewer: ```suggestion
    - Optional: Open another terminal window and run `k9s`. Use the arrow keys to scroll down to the posthog clusters and keep an eye on this for the duration of the upgrade. [`k9s`](https://k9scli.io/) is a terminal GUI that makes it easier to manage and observe your deployed Kubernetes applications.
```

>     - On a separate PR, navigate to [`playground.yaml`](https://github.com/PostHog/vpc/blob/main/client_values/posthog/playground.yaml) and replace the `image: -> tag:` value with the `release-[version]-unstable` tag found in Docker Hub. Tag the previous release manager on the PR and have it merged to `master`.

Reviewer: ```suggestion
    - On a separate PR, navigate to [`playground.yaml`](https://github.com/PostHog/vpc/blob/main/client_values/posthog/playground.yaml) and replace the `image: -> tag:` value with the `release-[version]-unstable` tag found in Docker Hub. Tag the previous release manager on the PR and have it merged to `master`.
        - ⚠️ Note that you might need to follow major upgrade notes as mentioned in the [upgrade guide](https://posthog.com/docs/self-host/deploy/digital-ocean#upgrading-the-chart), the same way our users would be required to, if so make any additional changes needed to the values file.
```

    - Copy the url of the new `playground.yaml` file. You can get that by navigating to the file [here](https://github.com/PostHog/vpc/blob/main/client_values/posthog/playground.yaml), clicking Raw in the Github UI, and copying the URL of that page.

  ![PostHog - Github Raw](../../images/05/release_playground_raw_github.png)

  ![PostHog - Github Raw File](../../images/05/release_playground_raw_file.png)
  
    - In a separate terminal window, follow the upgrade instructions [here](https://posthog.com/docs/self-host/deploy/digital-ocean#upgrading-the-chart). Replace `values.yaml` in the last upgrade command with the url you copied in the previous step. Example:

  ```shell
    helm upgrade -f https://raw.githubusercontent.com/PostHog/vpc/main/client_values/posthog/playground.yaml?token=ABC --timeout 20m --namespace posthog posthog posthog/posthog --atomic --wait --wait-for-jobs --debug
  ```

>     - If there are any failures showing up in `k9s` during the upgrade, ask `team-platform` for guidance.

Reviewer: ```suggestion
    - Optional: Keep an eye on the progress of the upgrade in `k9s`
    - If the `helm upgrade` command fails or if in the end the output for `kubectl get pods -n posthog` doesn't show everything as running, then ask `team-platform` for guidance.
```

    - Go to the [playground](https://playground.posthog.net/) and test that everything is working as expected. Check that the version running is the same as the one we're releasing.

> > ⚠️ Note that you might need to follow major upgrade notes as mentioned in the upgrade guide, the same way our users would be required to.

Reviewer: ```suggestion
```
